src/main/python/city_specific_analysis/nyc_analysis.py

```diff
@@ -0,0 +1,1146 @@
+
+    def num_people(row):
+        mode = row['mode']
+        if mode in ['walk', 'bike']:
+            return 1
+        elif mode == 'car':
+            return 1 + row['numPassengers']
+        else:
+            return row['numPassengers']
+
+    def benchmark():
+        data = """mode,Entering,Leaving
+subway,2241712,2241712
+car,877978,877978
+bus,279735,279735
+rail,338449,338449
+ferry,66932,66932
+bike,33634,33634
+tram,3528,3528
+        """
+        return pd.read_csv(StringIO(data)).set_index('mode')
+
+    f = df[(df['type'] == 'PathTraversal')][['mode', 'numPassengers', 'startX', 'startY', 'endX', 'endY']].copy(
+        deep=True)
+
+    f['numPeople'] = f.apply(lambda row: num_people(row), axis=1)
+    f = f[f['numPeople'] > 0]
+
+    f['startIn'] = f.apply(lambda row: inside(row['startX'], row['startY']), axis=1)
+    f['endIn'] = f.apply(lambda row: inside(row['endX'], row['endY']), axis=1)
+    f['numIn'] = f.apply(lambda row: row['numPeople'] if not row['startIn'] and row['endIn'] else 0, axis=1)
+
+    s = f.groupby('mode')[['numIn']].sum()
+    b = benchmark()
+
+    t = pd.concat([s, b], axis=1)
+    t.fillna(0, inplace=True)
+
+    t['percentIn'] = t['numIn'] * 100 / t['numIn'].sum()
+    t['percent_ref'] = t['Entering'] * 100 / t['Entering'].sum()
+
+    t = t[['numIn', 'Entering', 'percentIn', 'percent_ref']]
+
+    t['diff'] = t['percentIn'] - t['percent_ref']
+    t['diff'].plot(kind='bar', title="Diff: current - reference, %", figsize=(7, 5), legend=False, fontsize=12)
+
+    t.loc["Total"] = t.sum()
+    return t
+
+
+def get_people_in(df):
+    polygon = Polygon([
+        (-74.005088, 40.779100),
+        (-74.034957, 40.680314),
+        (-73.968867, 40.717604),
+        (-73.957924, 40.759091)
+    ])
+
+    def inside(x, y):
+        point = Point(x, y)
+        return polygon.contains(point)
+
+    def num_people(row):
+        mode = row['mode']
+        if mode in ['walk', 'bike']:
+            return 1
+        elif mode == 'car':
+            return 1 + row['numPassengers']
+        else:
+            return row['numPassengers']
+
+    f = df[(df['type'] == 'PathTraversal') & (df['mode'].isin(['car', 'bus', 'subway']))][
+        ['mode', 'numPassengers', 'startX', 'startY', 'endX', 'endY']].copy(deep=True)
+
+    f['numPeople'] = f.apply(lambda row: num_people(row), axis=1)
+    f = f[f['numPeople'] > 0]
+
+    f['startIn'] = f.apply(lambda row: inside(row['startX'], row['startY']), axis=1)
+    f['endIn'] = f.apply(lambda row: inside(row['endX'], row['endY']), axis=1)
+    f['numIn'] = f.apply(lambda row: row['numPeople'] if not row['startIn'] and row['endIn'] else 0, axis=1)
+
+    s = f.groupby('mode')[['numIn']].sum()
+
+    s.fillna(0, inplace=True)
+
+    s['percentIn'] = s['numIn'] * 100 / s['numIn'].sum()
+
+    return s['percentIn']
+
+
+def diff_people_in(current, base):
+    def reference():
+        data = """date,subway,bus,car
+07/05/2020,-77.8,-35,-21.8
```

Comment:
There are a number of hard-coded values like this that make me question it. If this is fine over time still then disregard - but I wanted to flag it at least.

Reply:
I believe that for future usage in the analysis of other cities all these NYC-specific python scripts will require changes anyway, so, we should be pretty safe.